.buildkite/pipeline.yml

```diff
@@ -23,7 +23,12 @@ steps:
         env:
           DRA_WORKFLOW: "snapshot"
       - label: ":package: DRA Publish Snapshot"
-        # publish runs in dry-run mode when triggered from PRs, see .buildkite/scripts/dra-publish.sh
+        if: |
+          // Only run when triggered from Unified Release and not from PRs
+          (
+            ( build.branch == 'main' || build.branch =~ /^[0-9]+\.[0-9]+\$/ ) &&
+            build.env("BUILDKITE_TRIGGERED_FROM_BUILD_PIPELINE_SLUG") == "unified-release-snapshot"
```

Comment:
This won't work for BCs we either need to add it as a condition or maybe just condition it on the existence of the env var and not it's value ?

Reply:
Good to know. We already do check for the same conditional earlier above: https://github.com/elastic/elastic-stack-installers/pull/255/files#diff-a3991ebf1475eb82acab13946ffc1eca02b43917f6d5bec3898f45c8b0b9bd53L12 (but also allow to run if it's a PR).

What would be the right value here to include BCs?

Reply:
It's `unified-release-staging` and then we'll have early agent release as well ... how about just `build.env("BUILDKITE_TRIGGERED_FROM_BUILD_PIPELINE_SLUG") != ""` ? The other condition will have to change as well

Reply:
> how about just `build.env("BUILDKITE_TRIGGERED_FROM_BUILD_PIPELINE_SLUG") != ""` ? The other condition will have to change as well

unified-release-staging is handled via a different group see https://github.com/elastic/elastic-stack-installers/pull/255/files#diff-a3991ebf1475eb82acab13946ffc1eca02b43917f6d5bec3898f45c8b0b9bd53L38

so based on that, I think we can be more specific with snapshot?

Reply:
My bad, did not look at enough context